### PR TITLE
build system: remove -x option from rimage invocation

### PR DESCRIPTION
**Pull request by mrajwa**
This patch removes the unnecessary and unscalable -x option.
This option aims to add an offset to the .data section of
its parent .module section. This is horrible idea because this
offset will vary depend on the compiler used. It is better and
much easier to just store only the .data section so the offset
is always 0.

Rimage side change -> https://github.com/thesofproject/rimage/pull/68

Signed-off-by: Marcin Rajwa <marcin.rajwa@linux.intel.com>

### CODE REVIEW

**Comment by mrajwa** [comment]
Some TGL issue, will debug in a moment...

**Comment by lgirdwood** [comment]
> Some TGL issue, will debug in a moment...

ok thanks., Btw, this PR should be to update the rimage submodule and not a local rimage commit..

**Comment by mrajwa** [comment]
> > Some TGL issue, will debug in a moment...
> 
> ok thanks., Btw, this PR should be to update the rimage submodule and not a local rimage commit..

I think the title confused you - I edit src/arch/xtensa/CMakeLists.txt here only. Will rephrase the title.

**Comment by mrajwa** [comment]
Please note that this option, while looking innocent is a lurking nightmare for any toolschain/compiler change!! And we are going to have some changes in this regards in the near feature. (clang is reaching out)

**Review by lgirdwood — APPROVED** [review]

**Comment by marc-hb** [comment]
> > ok thanks., Btw, this PR should be to update the rimage submodule and not a local rimage commit..
> ​and we have no way to to test this in CI - so this seems the best approach for validation.

Can you elaborate @lgirdwood ? I believe can test  in CI submodule updates before merge just fine (plus github has awesome support for submodules, it pains me to admit that)

(and yes I know this PR was not about submodules, sorry about the submodule digression)

**Comment by lgirdwood** [comment]
> > > ok thanks., Btw, this PR should be to update the rimage submodule and not a local rimage commit..
> > > ​and we have no way to to test this in CI - so this seems the best approach for validation.
> 
> Can you elaborate @lgirdwood ? I believe can test in CI submodule updates before merge just fine (plus github has awesome support for submodules, it pains me to admit that)
> 
> (and yes I know this PR was not about submodules, sorry about the submodule digression)

I was meaning "how do we test rimage PRs will build a viable FW image in the rimage CI".

**Comment by lgirdwood** [comment]
@marc-hb ok, we need this in the Readme.md for rimage. Please state the flows that developers should use.We obviously don't want maintainers to merge in SOF, so we need a way to state this and for it to be obvious that this is for CI only. Thanks !

**Comment by marc-hb** [comment]
> @marc-hb ok, we need this in the Readme.md for rimage. Please state the flows that developers should use.

Done (and tested) https://github.com/thesofproject/rimage/pull/71

**Pull request merged**
